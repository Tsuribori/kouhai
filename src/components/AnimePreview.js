import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import { Card, CardActionArea, CardActions, CardContent, CardMedia, Button, Typography, Grid } from '@material-ui/core';

const styles = theme => ({
  image: {
    paddingTop: theme.spacing.unit * 3,
  }
});
  
function AnimePreview(props) {
  return (
    <Grid item
      xs={12}
      sm={6}
      md={3}
      lg={2}
    >
      <Card>
        <CardActionArea>
          <div className={props.classes.image}>
            <Grid
              container
              justify="center"
            >
              <Grid
                item
                xs={10}
              >
                <CardMedia 
                  component="img"
                  image={props.image_url}
                  title={props.title}
                />
              </Grid>
            </Grid>
          </div>
          <CardContent>
            <Typography gutterBottom variant="h5" component="h3">
              {props.title} 
            </Typography>
          </CardContent>
        </CardActionArea>
        <CardActions>
          <Button
            size="small"
            color="primary"
            component={Link}
            to={`/anime/${props.mal_id}`}
          >
            More
          </Button>
        </CardActions>
      </Card>
    </Grid>
  );
}

AnimePreview.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(AnimePreview);
