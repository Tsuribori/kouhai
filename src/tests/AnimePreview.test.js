import React from 'react';
import './TestSetUp.js';
import AnimePreview from '../components/AnimePreview.js';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const path = require('path');
const fs = require('fs');

describe('AnimePreview test', () => {
  test('Test AnimePreview snapshot', () => {
    const data = JSON.parse(fs.readFileSync(path.resolve(__dirname, './Detail.json')));
    const tree = shallow(<AnimePreview {...data} />).dive();
    expect(toJson(tree)).toMatchSnapshot();
  });
}); 
