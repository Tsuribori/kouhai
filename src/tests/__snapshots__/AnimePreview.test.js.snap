// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnimePreview test Test AnimePreview snapshot 1`] = `
<WithStyles(Grid)
  item={true}
  lg={2}
  md={3}
  sm={6}
  xs={12}
>
  <WithStyles(Card)>
    <WithStyles(CardActionArea)>
      <div
        className="AnimePreview-image-1"
      >
        <WithStyles(Grid)
          container={true}
          justify="center"
        >
          <WithStyles(Grid)
            item={true}
            xs={10}
          >
            <WithStyles(CardMedia)
              component="img"
              image="https://cdn.myanimelist.net/images/anime/4/19644.jpg"
              title="Cowboy Bebop"
            />
          </WithStyles(Grid)>
        </WithStyles(Grid)>
      </div>
      <WithStyles(CardContent)>
        <WithStyles(Typography)
          component="h3"
          gutterBottom={true}
          variant="h5"
        >
          Cowboy Bebop
        </WithStyles(Typography)>
      </WithStyles(CardContent)>
    </WithStyles(CardActionArea)>
    <WithStyles(CardActions)>
      <WithStyles(Button)
        color="primary"
        component={[Function]}
        size="small"
        to="/anime/1"
      >
        More
      </WithStyles(Button)>
    </WithStyles(CardActions)>
  </WithStyles(Card)>
</WithStyles(Grid)>
`;
